src/componentWithName.js

```diff
@@ -16,7 +16,16 @@ function hasName(name, prop, propName, componentName) {
   }
 
   const { type } = prop;
-  if (getComponentName(type) !== name) {
+  const componentNameFromType = getComponentName(type);
+
+  const isRegex = name instanceof RegExp;
```

Comment:
`instanceof` isn't reliable - please use https://www.npmjs.com/package/is-regex instead :-)